os/net/netdev/netdev_ioctl.c

```diff
@@ -1067,6 +1066,39 @@ int lwip_func_ioctl(FAR struct socket *sock, int cmd, void  *arg)
 
 	return ret;
 }
+
+/****************************************************************************
+ * Function: lwipioctl
+ *
+ * Description:
+ *   Call lwip_ioctl() with FIONREAD/FIONBIO commands or
+ *   call lwip API with SIOCLWIP command
+ *
+ * Parameters:
+ *   sockfd   Socket file descriptor
+ *   cmd      The ioctl command
+ *   arg      Type of the information to get
+ *
+ * Returned Value:
+ *   0 on success, negated errno on failure.
+ *
+ ****************************************************************************/
+
+int lwipioctl(int sockfd, int cmd, void *arg)
+{
+	int ret = -ENOTTY;
+
+	if (cmd == FIONREAD || cmd == FIONBIO) {
+		ret = lwip_ioctl(sockfd, (long)cmd, arg);
+		if (ret == -1) {
+			ret = -get_errno();
+		}
+	} else if (cmd == SIOCLWIP) {
+		return lwip_func_ioctl(cmd, arg);
+	}
+
+	return ret;
```

Comment:
ret is not neccessary here,
you can return -get_errno() & lwip_func_ioctl directly.
Some tool will verify this as an defect(low level) because after 'else if' state you do not use ret.

Reply:
Thank you I'll fix.